### PR TITLE
Bug 1789462: fix OAuth discovery in IPv6 environments

### PR DESCRIPTION
**Pull request by stlaz**

### CODE REVIEW

**Comment by stlaz** [comment]
/retest
unrelated errors, oauth-proxy seems to be running fine

**Comment by russellb** [comment]
@stlaz can you help make sure #150 merges once 4.3.z is open for backports again?

File: providers/openshift/provider.go [hunk 1]
```diff
@@ -633,6 +633,10 @@ func getKubeAPIURLWithPath(path string) *url.URL {
 	}
 
 	if host := os.Getenv("KUBERNETES_SERVICE_HOST"); len(host) > 0 {
+		// assume IPv6 if host contains colons
+		if strings.IndexByte(host, ':') != -1 {
+			host = "[" + host + "]"
+		}
```
Comment (russellb) [inline providers/openshift/provider.go]: a safer check would be to parse this with `net.ParseIP()` and then check if `ip.To4()` succeeds or not

Comment (russellb) [reply providers/openshift/provider.go]: or likely even better, wherever this later gets used in a URL, use `net.JoinHostPort()` which will automatically add brackets around an IPv6 address if needed

**Pull request merged**